### ISSUE TITLE
Extensions not installing on vscode insiders when running inside a dev container

### ISSUE THREAD

**Issue by kurt-mueller-osumc** [microsoft/vscode#175705]
Does this issue occur when all extensions are disabled?: n/a

n/a I think. The code editor cannot install extensions when booting up a development container. I've ran the 'Help: Start Extension Bisect' command to reload the container without any extensions installed.... unfortunately, when I attempt to install the `ms-python.python` extension, the extension is stuck on 'Installing'.

- VS Code Version: 

Version: 1.76.0-insider
Commit: 92da9481c0904c6adfe372c12da3b7748d74bdcb
Date: 2023-02-28T11:50:50.947Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Darwin arm64 22.3.0
Sandboxed: Yes

- OS Version: macOS Ventura 13.2.1

Notes:

I'm on an macos and am using rosetta to emulate `x86_64` architecture.

**Comment by sandy081** [comment]
Can you please share the following log while you reproduce the issue?

- F1 > Open View... > Remote Server

**Comment by twanwv** [comment]
Logging in 1.76.0
`2023-03-02 15:31:50.298 [info] 




2023-03-02 15:31:50.358 [info] Extension host agent started.
2023-03-02 15:31:50.392 [debug] Removing uninstalled extensions: []
2023-03-02 15:31:50.495 [info] [127.0.0.1][eafae52d][ManagementConnection] New connection established.
2023-03-02 15:31:50.676 [info] [127.0.0.1][b5ea7270][ExtensionHostConnection] New connection established.
2023-03-02 15:31:50.703 [info] [127.0.0.1][b5ea7270][ExtensionHostConnection] <209> Launched Extension Host Process.
2023-03-02 15:32:24.559 [info] [127.0.0.1][eafae52d][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.
2023-03-02 15:32:24.623 [info] [127.0.0.1][b5ea7270][ExtensionHostConnection] <209> Extension Host Process exited with code: 0, signal: null.
2023-03-02 15:32:27.697 [info] [127.0.0.1][c7784cbf][ManagementConnection] New connection established.
2023-03-02 15:32:28.398 [info] [127.0.0.1][8d893acd][ExtensionHostConnection] New connection established.
2023-03-02 15:32:28.406 [info] [127.0.0.1][8d893acd][ExtensionHostConnection] <797> Launched Extension Host Process.
`

Logging in 1.75.1
`2023-03-02 15:33:52.356 [info] 




2023-03-02 15:33:52.406 [info] Extension host agent started.
2023-03-02 15:33:52.514 [info] [127.0.0.1][36eba529][ManagementConnection] New connection established.
2023-03-02 15:33:52.516 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.15]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0-insider.
2023-03-02 15:33:52.576 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-2023.2.1000592019]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0.
2023-03-02 15:33:52.622 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.15]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0-insider.
2023-03-02 15:33:52.655 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-2023.2.1000592019]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0.
2023-03-02 15:33:52.793 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.15]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0-insider.
2023-03-02 15:33:52.798 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.15]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0-insider.
2023-03-02 15:33:52.806 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.15]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0-insider.
2023-03-02 15:33:52.808 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.15]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0-insider.
2023-03-02 15:33:52.810 [info] [127.0.0.1][147a00d1][ExtensionHostConnection] New connection established.
2023-03-02 15:33:52.823 [info] [127.0.0.1][147a00d1][ExtensionHostConnection] <236> Launched Extension Host Process.
2023-03-02 15:33:52.858 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-2023.2.1000592019]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0.
2023-03-02 15:33:52.873 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-2023.2.1000592019]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0.
2023-03-02 15:33:52.880 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-2023.2.1000592019]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0.
2023-03-02 15:33:52.886 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-2023.2.1000592019]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0.
2023-03-02 15:33:52.935 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.15]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0-insider.
2023-03-02 15:33:53.139 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-2023.2.1000592019]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0.
2023-03-02 15:33:53.237 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.15]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0-insider.
2023-03-02 15:33:53.241 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.15]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0-insider.
2023-03-02 15:33:53.242 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.15]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0-insider.
2023-03-02 15:33:53.243 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.15]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0-insider.
2023-03-02 15:33:53.257 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-2023.2.1000592019]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0.
2023-03-02 15:33:53.263 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-2023.2.1000592019]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0.
2023-03-02 15:33:53.267 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-2023.2.1000592019]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0.
2023-03-02 15:33:53.269 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-2023.2.1000592019]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0.
2023-03-02 15:33:54.305 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.15]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0-insider.
2023-03-02 15:33:54.314 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-2023.2.1000592019]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0.
2023-03-02 15:33:59.435 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-renderers-1.0.15]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0-insider.
2023-03-02 15:33:59.439 [error] [/root/.vscode-server/extensions/ms-toolsai.jupyter-2023.2.1000592019]: Extension is not compatible with Code 1.75.1. Extension requires: ^1.76.0.`

**Comment by kurt-mueller-osumc** [comment]
Hi Sandeep,

There is no option to `View Remote Server` but I do have the ability to `View Server`. I've taken a screenshot of the command I ran:

<img width="1728" alt="Screenshot 2023-03-02 at 10 30 52 AM" src="https://user-images.githubusercontent.com/66080261/222475043-0d2e3b63-c693-4e0d-8024-28009e4ff117.png">

It gave me the following output:

```
2023-03-02 15:29:03.775 [info] 

2023-03-02 15:29:03.879 [info] Extension host agent started.
2023-03-02 15:29:03.997 [debug] Removing uninstalled extensions: []
2023-03-02 15:29:05.651 [info] [127.0.0.1][980f0093][ManagementConnection] New connection established.
2023-03-02 15:29:06.440 [info] [127.0.0.1][bab05b13][ExtensionHostConnection] New connection established.
2023-03-02 15:29:06.474 [info] [127.0.0.1][bab05b13][ExtensionHostConnection] <609> Launched Extension Host Process.
```

Let me know if you need any adition info.

**Comment by kurt-mueller-osumc** [comment]
I think extensions fail to install on containers that emulate x86 architecture using Rosetta virtualization. I've been able to install extensions in containers that are `aarch` based.

**Comment by sandy081** [comment]
I do not see any installation related entries in the logs. 
May I know how are you installing the extension? Are you triggering this from extensions UI? 
Can you also please enable log level to trace `F1 > Set Log Level... > Trace` and reproduce the issue?

**Comment by sandy081** [comment]
@chrmarti It seems this is specific to containers that emulate x86 architecture using Rosetta virtualization - does it rings any bells?

**Comment by kurt-mueller-osumc** [comment]
@sandy081 When I set the log levels to "trace", I can see a `PackageIsUnreadable` exception being thrown when I attempt to install extensions (via vscode's extension ui):

```
[45916 ms] [15:54:20] Extension signature verification details for ms-toolsai.jupyter-renderers 1.0.15:
Package file size:  82
Signature archive file size:  11100
Manifest file size:  6076
Signature file size:  11986
Manifest file hash:  iNySwbbf5cnq+zEvmBtV3PPIo/Zw518d02NsfAKeoZQ=
Signature file hash:  gG/0OxuEf7CddE+i89TBtc0oXH7DR2lNo2V9hwI/G7A=
System.IO.FileNotFoundException: IO_FileNotFound_FileName, /home/.vscode-server-insiders/extensionsCache/ms-toolsai.jupyter-renderers-1.0.15
IO_FileName_Name, /home/.vscode-server-insiders/extensionsCache/ms-toolsai.jupyter-renderers-1.0.15
   at Interop.ThrowExceptionForIoErrno(ErrorInfo, String, Boolean)
   at Interop.CheckIo(Error, String , Boolean )
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, OpenFlags, Int32, Func`4)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, FileMode, FileAccess, FileShare, FileOptions, Int64, UnixFileMode, Int64& , UnixFileMode& , Func`4 )
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1 , Func`4 )
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.UnixFileStreamStrategy..ctor(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream, String, FileMode, FileAccess, FileShare, Int32, FileOptions, Int64, Nullable`1)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, FileOptions, Int64)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, FileOptions)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, Boolean)
   at System.IO.FileInfo.OpenRead()
   at Microsoft.VisualStudio.Extensions.Signing.ExtensionPackage.OpenRead(FileInfo)
   at Microsoft.VisualStudio.Extensions.Signing.CLI.Program.VerifyAsync(FileInfo, FileInfo, FileInfo, FileInfo, Func`2, Boolean, Boolean)
Exit code:  PackageIsUnreadable

[45927 ms] [15:54:20] Extension signature verification details for ms-toolsai.jupyter-keymap 1.1.0:
Package file size:  78
Signature archive file size:  9554
Manifest file size:  1266
Signature file size:  11977
Manifest file hash:  XLhYoMpV96SN8l4cK7gR2SGGczuL3n+zsMNkEZGS3So=
Signature file hash:  h7qtOCYh5W21g6iWYrAf8DCNIBGYtg5kBPbUZzm08n8=
System.IO.FileNotFoundException: IO_FileNotFound_FileName, /home/.vscode-server-insiders/extensionsCache/ms-toolsai.jupyter-keymap-1.1.0
IO_FileName_Name, /home/.vscode-server-insiders/extensionsCache/ms-toolsai.jupyter-keymap-1.1.0
   at Interop.ThrowExceptionForIoErrno(ErrorInfo, String, Boolean)
   at Interop.CheckIo(Error, String , Boolean )
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, OpenFlags, Int32, Func`4)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, FileMode, FileAccess, FileShare, FileOptions, Int64, UnixFileMode, Int64& , UnixFileMode& , Func`4 )
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1 , Func`4 )
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.UnixFileStreamStrategy..ctor(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream, String, FileMode, FileAccess, FileShare, Int32, FileOptions, Int64, Nullable`1)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, FileOptions, Int64)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, FileOptions)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, Boolean)
   at System.IO.FileInfo.OpenRead()
   at Microsoft.VisualStudio.Extensions.Signing.ExtensionPackage.OpenRead(FileInfo)
   at Microsoft.VisualStudio.Extensions.Signing.CLI.Program.VerifyAsync(FileInfo, FileInfo, FileInfo, FileInfo, Func`2, Boolean, Boolean)
Exit code:  PackageIsUnreadable

[45942 ms] [15:54:20] Extension signature verification details for ms-toolsai.vscode-jupyter-slideshow 0.1.5:
Package file size:  88
Signature archive file size:  10219
Manifest file size:  2789
Signature file size:  11978
Manifest file hash:  AqePx/Jq+/h0mm3d80rh1MJSKc78Xq3ERcVCC+CDPhQ=
Signature file hash:  yglIqMdlccgWMean0TXtRRiD+5RejR1LWNkR7gWAIZA=
System.IO.FileNotFoundException: IO_FileNotFound_FileName, /home/.vscode-server-insiders/extensionsCache/ms-toolsai.vscode-jupyter-slideshow-0.1.5
IO_FileName_Name, /home/.vscode-server-insiders/extensionsCache/ms-toolsai.vscode-jupyter-slideshow-0.1.5
   at Interop.ThrowExceptionForIoErrno(ErrorInfo, String, Boolean)
   at Interop.CheckIo(Error, String , Boolean )
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, OpenFlags, Int32, Func`4)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, FileMode, FileAccess, FileShare, FileOptions, Int64, UnixFileMode, Int64& , UnixFileMode& , Func`4 )
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1 , Func`4 )
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.UnixFileStreamStrategy..ctor(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream, String, FileMode, FileAccess, FileShare, Int32, FileOptions, Int64, Nullable`1)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, FileOptions, Int64)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, FileOptions)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, Boolean)
   at System.IO.FileInfo.OpenRead()
   at Microsoft.VisualStudio.Extensions.Signing.ExtensionPackage.OpenRead(FileInfo)
   at Microsoft.VisualStudio.Extensions.Signing.CLI.Program.VerifyAsync(FileInfo, FileInfo, FileInfo, FileInfo, Func`2, Boolean, Boolean)
Exit code:  PackageIsUnreadable

[45984 ms] [15:54:20] Extension signature verification details for ms-toolsai.jupyter 2023.2.1000592019:
Package file size:  83
Signature archive file size:  17353
Manifest file size:  24441
Signature file size:  12017
Manifest file hash:  16dnVUBrhyTjSMrCk1/ZR8Tdb/pfNoJ+F19PWP5MU5Y=
Signature file hash:  EudiXyv1VScg+MtaveqbbAos/OP8bOnscTngAdYU67E=
System.IO.FileNotFoundException: IO_FileNotFound_FileName, /home/.vscode-server-insiders/extensionsCache/ms-toolsai.jupyter-2023.2.1000592019
IO_FileName_Name, /home/.vscode-server-insiders/extensionsCache/ms-toolsai.jupyter-2023.2.1000592019
   at Interop.ThrowExceptionForIoErrno(ErrorInfo, String, Boolean)
   at Interop.CheckIo(Error, String , Boolean )
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, OpenFlags, Int32, Func`4)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, FileMode, FileAccess, FileShare, FileOptions, Int64, UnixFileMode, Int64& , UnixFileMode& , Func`4 )
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1 , Func`4 )
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.UnixFileStreamStrategy..ctor(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream, String, FileMode, FileAccess, FileShare, Int32, FileOptions, Int64, Nullable`1)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, FileOptions, Int64)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, FileOptions)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, Boolean)
   at System.IO.FileInfo.OpenRead()
   at Microsoft.VisualStudio.Extensions.Signing.ExtensionPackage.OpenRead(FileInfo)
   at Microsoft.VisualStudio.Extensions.Signing.CLI.Program.VerifyAsync(FileInfo, FileInfo, FileInfo, FileInfo, Func`2, Boolean, Boolean)
Exit code:  PackageIsUnreadable

[45994 ms] [15:54:20] Extension signature verification details for ms-toolsai.vscode-jupyter-cell-tags 0.1.6:
Package file size:  88
Signature archive file size:  10628
Manifest file size:  3690
Signature file size:  12019
Manifest file hash:  unCeiJeg1jTT/MymV3D5FyXMnKJ6elXaKo1BLh3j9wo=
Signature file hash:  5UaTECA1TDNT/YomULDUHqrCXIpItcO0DDVFKhekV2A=
System.IO.FileNotFoundException: IO_FileNotFound_FileName, /home/.vscode-server-insiders/extensionsCache/ms-toolsai.vscode-jupyter-cell-tags-0.1.6
IO_FileName_Name, /home/.vscode-server-insiders/extensionsCache/ms-toolsai.vscode-jupyter-cell-tags-0.1.6
   at Interop.ThrowExceptionForIoErrno(ErrorInfo, String, Boolean)
   at Interop.CheckIo(Error, String , Boolean )
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, OpenFlags, Int32, Func`4)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, FileMode, FileAccess, FileShare, FileOptions, Int64, UnixFileMode, Int64& , UnixFileMode& , Func`4 )
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1 , Func`4 )
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.UnixFileStreamStrategy..ctor(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String, FileMode, FileAccess, FileShare, FileOptions, Int64, Nullable`1)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream, String, FileMode, FileAccess, FileShare, Int32, FileOptions, Int64, Nullable`1)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, FileOptions, Int64)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, FileOptions)
   at System.IO.FileStream..ctor(String, FileMode, FileAccess, FileShare, Int32, Boolean)
   at System.IO.FileInfo.OpenRead()
   at Microsoft.VisualStudio.Extensions.Signing.ExtensionPackage.OpenRead(FileInfo)
   at Microsoft.VisualStudio.Extensions.Signing.CLI.Program.VerifyAsync(FileInfo, FileInfo, FileInfo, FileInfo, Func`2, Boolean, Boolean)
Exit code:  PackageIsUnreadable

[77751 ms] [IPC Library: Pty Host] TRACE ptyService#setLayoutInfo [ { isActive: true, activePersistentProcessId: 0, terminals: [] } ]
```

**Comment by sandy081** [comment]
Seems extension signing verification related.

CC @isidorn / @dtivel

**Comment by RaymondLChastain** [comment]
I am also experiencing this issue.

Attempting to manually install in the container _sometimes_ fixes the extensions but for some extensions - seemingly chosen at random - you'll get PackageIsUnreadable errors preventing installation.

**Comment by kf6kjg** [comment]
Since https://github.com/microsoft/vscode-remote-release/issues/8169 has been closed as a duplicate of this issue, I figured that I should note here that the closed issue has a lot of commentary and debugging of this issue.

**Comment by 1995parham** [comment]
I don't have this issue using Insider version on M2Pro, but after upgrading macOS and Insider I have this issue and disable the plugin verification resolved the issue for now.

**Comment by sandy081** [comment]
Can you (those who subscribed) please verify if this issue still exists?

- Remove the setting `extensions.verifySignature` if you have set it (from user and remote settings)
- Restart VS Code
- Set Log Level to Trace - `F1 > Developer: Set Log Level > Trace`
- Install an extension
- Share the output - `F1 > Show View... > Shared`

**Comment by mdrakiburrahman** [comment]
@sandy081 - I just reprod it 2 days ago. But my version of VSCode might not be the version you've pushed the fix to.

This workaround unblocked our team:
![image](https://github.com/microsoft/vscode/assets/46581776/92103d9f-8e37-486f-beeb-cda7dc235cd4)


Could you let me know what to try:
1. Which VSCode, regular, or insiders
2. Which version of VSCode to install to test the fix

**Comment by sandy081** [comment]
@mdrakiburrahman Thanks for reaching out. It would be great if you can update to latest stable and try. Can you please let us know with which extension you are having this issue? Can you please share your platform and following log while you reproduce

- F1 > Open View... > Shared

**Comment by mdrakiburrahman** [comment]
Hi @sandy081,

Looks like it's working now - I might have just received the update last night!

![image](https://github.com/microsoft/vscode/assets/46581776/501963b3-2353-4ca1-ac62-c0d6ef050645)

The way I noticed this one my daily used extensions, [Metals](https://scalameta.org/metals/docs/editors/vscode/) no longer installed in our devcontainer.

It's now installing fine:

![image](https://github.com/microsoft/vscode/assets/46581776/a8b90e64-213d-4ced-b555-311f77914714)

Thank you!

**Comment by ragheb-deriv** [comment]
I just reported the same issue here : https://github.com/microsoft/vscode-remote-release/issues/9525

**Comment by Wesztman** [comment]
I can also confirm that I had problems with extensions not installing in stable 

![image](https://github.com/microsoft/vscode/assets/54413402/0bcdc7e2-2adf-4ca2-8aea-1d0b1db80c55)

But with insiders it is now working :)

![image](https://github.com/microsoft/vscode/assets/54413402/bc79b5fa-6987-4520-8e4b-642f9b12f812)

**Comment by humphd** [comment]
I've just updated to 1.87.0 and it happened to me.  None of the extensions listed in my `.devcontainer/devcontainer.json` are installed now.  I tried the suggestion in https://github.com/microsoft/vscode/issues/175705#issuecomment-1906509931, but it didn't have any effect.

<img width="386" alt="Screenshot 2024-03-05 at 3 22 06 PM" src="https://github.com/microsoft/vscode/assets/427398/8dea7fc3-32a0-4586-b5eb-962f1a133b31">

Version: 1.87.0 (Universal)
Commit: 019f4d1419fbc8219a181fab7892ebccf7ee29a2
Date: 2024-02-27T23:42:56.944Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 23.2.0

**Comment by trollspank** [comment]
Having the exact same issue on Mac M1 as well, I've also tried installing the non-M1 version of VSCode, same issue. Wiped all my extensions, restarted, nothing (except errors trying to start my devcontainer w/o the docker extension of course)

**Comment by mdrakiburrahman** [comment]
Hi @sandy081

Bad news - this broke again on `1.87.1`:

![image](https://github.com/microsoft/vscode/assets/46581776/32a04508-69b7-4a1b-8ee7-85ea2a495223)

Workaround was to revert back `"extensions.verifySignature": false` again.

---

Is there regression tests setup for whoever is making these commits to break extensions?

**Comment by papakpmartin** [comment]
I'm also seeing this problem. Apple M2, running 13.3.1 (22E261).

`customizations.vscode.settings.extensions.verifySignature: false` allows things to work.

**Comment by DanseMacabre** [comment]
Is there any update on this? For reference to anyone else checking this thread, `"extensions.verifySignature": false` goes in the Remote Settings JSON file, and if your container is destroyed and re-built from scratch, you may need to re-add this line to the file as it exists on the container.

**Comment by j-bennet** [comment]
+1, Python extension, even with `"extensions.verifySignature": false`.

**Comment by bo-values** [comment]
My workaround is here.

--------

MacBook Air M2
macOS Venture 13.2.1
Visual Studio Code 1.89.0 (Universal)
DevContainers v0.362.0
Debian OS
````
## Dockerfile
FROM --platform=linux/amd64 mcr.microsoft.com/vscode/devcontainers/base:dev-bullseye
````
--------

1. set `"extensions.verifySignature": false` at devcontainer.json
2. build container > I faced extensions is stuck on 'Installing'.
3. then, click cloud button at extensions panel like image. some extensions were installed,but not perfect.
4. click `Rebuild Container`.
5. all extensions was installed!

![image](https://github.com/microsoft/vscode/assets/129837656/e5e0f1e5-38c5-4a98-9e9f-fdf029a81102)
![image](https://github.com/microsoft/vscode/assets/129837656/6aebe711-50c6-4389-bc51-171d1a0c7c21)

**Comment by coffepowered** [comment]
Experiencing the same issue on M1 pro: extensions don't install on the Remote Container

Host:
- osx 14.1.1
- vscode 1.89.1 arm64 via homebrew (briefly tried previous version as well)
- no devcontainer.json, just attaching to a container in docker-compose

My tests:
- Extensions install locally
- Extension install in attached container if I don't specify platform
- ms-python extension does NOT install in attached container if I build using platform: linux/amd64 (tried "extensions.verifySignature": false in .vscode/settings.json with no luck)

**Comment by DanseMacabre** [comment]
> Experiencing the same issue on M1 pro: extensions don't install on the Remote Container
> 
> 
> 
> Host:
> 
> - osx 14.1.1
> 
> - vscode 1.89.1 arm64 via homebrew (briefly tried previous version as well)
> 
> - no devcontainer.json, just attaching to a container in docker-compose
> 
> 
> 
> My tests:
> 
> - Extensions install locally
> 
> - Extension install in attached container if I don't specify platform
> 
> - ms-python extension does NOT install in attached container if I build using platform: linux/amd64 (tried "extensions.verifySignature": false in .vscode/settings.json with no luck)
> 
> 
> 
> 
> 
> 

Try following the workaround in my previous comment:

Add the line to the remote workspace json file. You may need to restart the container or kill the hung vscode process(es) that are in the container to have this change take effect.

**Comment by mindplay-dk** [comment]
I'm on Win 11 using WSL2 with Docker, stable VS Code `1.95.2` - no extensions get installed.

I get this warning for any extensions listed in my `devcontainer.json`:

> This extension is disabled in this workspace because it is defined to run in the Remote Extension Host. Please install the extension in 'Dev Container: chillproductions.com' to enable

![Image](https://github.com/user-attachments/assets/e456bb44-7d67-4fdc-ad79-5d03cbf67a1c)

I have to install the extensions again every time I open this project.

It sounds like some users have been able to work around this adding `"extensions.verifySignature": false` to `customizations.vscode.extensions` - this did not work for me.

Rebuilding the container, no difference.

Is there a recommended workaround? Should I move the extensions to suggestions instead? That way at least when someone else checks out the project, they can see which extensions were supposed to be installed.

If it matters, my `Dockerfile` derives from `mcr.microsoft.com/devcontainers/php:8.3`.

Let me know if there's anything I can test or provide?

**Comment by DanseMacabre** [comment]
Has there been a fix on this - my recent reset didn't require the usual workaround. Anyone else seeing this now behave?
